Fixes things to use new c# 7 features via help of Visual Studio 2017

chuggafan opened the pull request:
This is essentially just getting us up to using the c# 7 features that visual studio 2017 can detect, it can help with cleaner code and less lines to do the same work

CrustyJew commented on UnitTesting/UnitTest1.cs:
```diff
@@ -9,7 +9,7 @@ namespace UnitTesting
     public class UnitTest1
     {
         [TestMethod]
-        public void getSubreddit()
```
Don't believe this should have changed as it's a method name and should be capitalized

CrustyJew commented:
I'm not entirely sure how I feel about this `out` convention, what advantage does it have exactly? And isn't C# 7 not officially released yet?

chuggafan commented:
The out convention is essentially ability to declare variables inside of functions that have "out" instead of declaring them beforehand, also the capitalization of method names is what Microsoft suggests, *shrug*

pimanac commented:
c# 7 is still in preview if I'm not mistaken.

Are there any functional changes here or is it all syntactic sugar?

chuggafan commented:
It's just syntactic sugar, nothing more, same functions are produced IIRC

CrustyJew commented on RedditSharp/WebAgent.cs:
```diff
@@ -104,8 +104,7 @@ static WebAgent()
         {
-            Uri uri;
-            if (!Uri.TryCreate(url, UriKind.Absolute, out uri))
+            if (!Uri.TryCreate(url, UriKind.Absolute, out var uri))
```
I second this, the slightly more wordy version is much more readable to me and feels more specific/easier to follow.

CrustyJew commented on RedditSharp/ToolBoxUserNotes.cs:
```diff
@@ -51,17 +51,18 @@ public static IEnumerable<TBUserNote> GetUserNotes(IWebAgent webAgent, string su
-                        uNote.Timestamp = UnixTimeStamp.UnixTimeStampToDateTime(note["t"].Value<long>());
-                        uNote.Url = UnsquashLink(subName, note["l"].ValueOrDefault<string>());
-
+                        TBUserNote uNote = new TBUserNote()
```
This change I do like and would merge. The rest of the `out` syntax sugar I'm not a big fan of.

chuggafan commented on RedditSharp/Subreddit Information/SubredditImage.cs:
```diff
@@ -27,8 +27,7 @@ public class SubredditImage
             Url = new Uri(url);
             // Handle legacy image urls
             // http://thumbs.reddit.com/FULLNAME_NUMBER.png
-            int discarded;
```
Actually for this specific number this doesn't matter, if wildcards were implemented I would use it as it is only used to see if it's an integer or not, nothing else, so I won't change that

chuggafan commented:
I removed all relevant syntactical sugar and reverted it, except for the "discarded" variable as it would only add lines of code to an irrelevant variable that's only use is for seeing whether or not the string is an integer

chuggafan commented:
I removed HTMLAgilityPack dependency in the .nuspec file and also moved all the files to .NETFramework 4.6.2 when being compiled, all .NET framework 4.5 code is supported by .NET framwork 4.6.2, so there should be no issue with updating for anyone if this gets pulled in

CrustyJew commented:
Can you re-open this pull request to the 2.0 branch instead

chuggafan commented:
Done

CrustyJew commented:
thanks
